compiler/front/cli_reporter.nim

```diff
@@ -1469,6 +1469,10 @@ proc reportBody*(conf: ConfigRef, r: SemReport): string =
     of rsemUnexpectedYield:
       result = "'yield' only allowed in an iterator"
 
+    of rsemYieldExpectedTupleConstr:
+      result = "tuple constructor expected for iterator returning a tuple" &
```

Comment:
"Literal tuple constructor (x, y, z, ...) expected for 'yield' as iterator returns a tuple type which contains a view"

Additional information might include
- annotation showing which part of the tuple is a view type
- what was found instead of a tuple ("but found procedure with a tuple return value")

Rephrasing is optional, the main point is to (1) provide at least some simplistic hint to fix, (2) show the language construct (yield) which causes the local issue, (3) more precise description of what is needed (**literal** tuple if I understand the error message correctly)

Bullet points are just TODO for later, feel free to just copy-paste as literal TODO in the code, don't bother with this for now.